### ISSUE TITLE
Intents should also stop execution of higher priority listeners

### ISSUE THREAD

**Issue by Moehritz** [SpigotMC/BungeeCord#1491]
By registering an Intent, you can do blocking stuff in an async executor, but still change the events result.

But there could be another use case for this: Imagine having a permissions-plugin. It loads your groups and stuff async in the LoginEvent(Priority LOW) (PreLoginEvent is not possible due to missing UUID) and registers an intent for this. A second plugin, that requires information about the players permissions in the LoginEvent aswell, won´t get them, as the async permission-loading isn´t done yet.
So basically, registering an intent does not interrupt the execution of the following (higher priority) listeners.
I don´t know wheter this is intended or not, by I can´t think of an easy way around this (except some sort of additional custom "PermissionLoadedEvent")

**Comment by minecrafter** [comment]
In order to add this without causing lag, you'd have to substantially redesign `EventBus`. In this sample scenario you could check this in `PostLoginEvent` though.

**Comment by Moehritz** [comment]
I am aware of the complexity of this problem, yes. And PostLoginEvent is no option here either, as there are other plugins trying to access permissions in LoginEvent. Until now, it basically works by using the mentioned "PermissionLoadedEvent", but this does not work with plugins that you can´t modify (premium resources or something like that)
